File: finalAssignmentBackend/src/main/java/com/tutict/finalassignmentbackend/entity/BackupRestore.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("backup_restore")
public class BackupRestore implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;

    @TableId(value = "backup_id", type = IdType.AUTO)
    private Integer backupId;

    @TableField("backup_file_name")
    private String backupFileName;

    @TableField("backup_time")
    private LocalDateTime backupTime;

    @TableField("restore_time")
    private LocalDateTime restoreTime;

    @TableField("restore_status")
    private String restoreStatus;

    @TableField("remarks")
    private String remarks;
}

File: final_asssignment_backend_go/project/internal/domain/backup_restore.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// BackupRestore 表示 backup_restore 表的实体
type BackupRestore struct {
	BackupID       int            `gorm:"column:backup_id;primaryKey;autoIncrement" json:"backup_id"`
	BackupFileName string         `gorm:"column:backup_file_name" json:"backup_file_name"`
	BackupTime     time.Time      `gorm:"column:backup_time" json:"backup_time"`
	RestoreTime    time.Time      `gorm:"column:restore_time" json:"restore_time"`
	RestoreStatus  string         `gorm:"column:restore_status" json:"restore_status"`
	Remarks        string         `gorm:"column:remarks" json:"remarks"`
	DeletedAt      gorm.DeletedAt `gorm:"index"` // 软删除字段（可选，GORM 默认支持）
}

// TableName 指定数据库表名
func (BackupRestore) TableName() string {
	return "backup_restore"
}
